File: tests/Identifiers/VisualBasicIdentifiers/Constructions/TypeConstraints.vb
Public Class TypeConstraints1(Of T As IEnumerable(Of IEnumerable(Of Integer)))
End Class

Public Class TypeConstraints2(Of T As {IEnumerable(Of IEnumerable(Of Integer)), IDisposable})
End Class

Public Class TypeConstraints3(Of T As System.Exception)
End Class

Public Class TypeConstraints4(Of T As {System.Exception})
End Class

Public Class TypeConstraints5(Of T As Exception)
End Class

File: tests/Identifiers/CSharpIdentifiers/Constructions/TypeConstraints.cs
using System;
using System.Collections.Generic;

namespace CSharpIdentifiers.Constructions
{
    internal class TypeConstraints1<T> where T : IEnumerable<IEnumerable<int>>
    {
    }

    internal class TypeConstraints2<T> where T : Exception
    {
    }

    internal class TypeConstraints3<T> where T : System.Exception
    {
    }
}